### PR TITLE
BjQmh9rH set up continuous integration pipeline

### PR DESCRIPTION
**Pull request by samuel776**
#### What does this PR do?
   -Setup Continuous Integration & Quality Assurance pipeline
#### Description of Task to be completed?
 - configure testing environment with jest and enzyme
 - configure github actions for continuous
-configure heroku pipeline on heroku
-configure codeclimate
-coveralls for pr checking
#### How should this be manually tested?
- clone it to the local repo
- run npm install
- test the home page by running npm test
#### What are the relevant trello id?
[BjQmh9rH](https://trello.com/c/BjQmh9rH/27-setup-continuous-integration-quality-assurance-pipeline)

### CODE REVIEW

**Review by GuevaraMarius — COMMENTED** [review]
Good work Samuel, too many files changed. You can stop tracking the coverage directory.

File: src/setupTest.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import Enzyme from 'enzyme';
+import EnzymeAdapter from 'enzyme-adapter-react-16';
+
+Enzyme.configure({
+  adapter: new EnzymeAdapter(),
+  dasableLifecycleMethods: true
+});
```
Comment (GuevaraMarius) [inline src/setupTest.js]: Add a new line at the end of the file

File: src/tests/tests.test.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+import { shallow } from 'enzyme';
+import {App} from '../router/App';
+
+
+describe('home page', () =>{
+  it('should return the welcoming page', () =>{
+    const wrapper = shallow(<App/>)
+    expect(wrapper).toMatchSnapshot();
+  });
+});
```
Comment (GuevaraMarius) [inline src/tests/tests.test.js]: Same here

File: webpack.config.js [hunk 3]
```diff
@@ -17,27 +20,52 @@ module.exports = {
...
+      template: path.join(__dirname, "public", "index.html"),
+    }),
+  ],
+  resolve: {
+    modules: [path.join(__dirname, "src"), "node_modules"],
+    extensions: ["*", ".js", ".jsx"],
   },
-  plugins: [new HtmlWebpackPlugin({ template: './public/index.html' })]
-};
+};
```
Comment (GuevaraMarius) [inline webpack.config.js]: Add a new line here too.

File: jest.config.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+  coverageDirectory: "./coverage",
+  moduleNameMapper: {
+    "\\.(css|less|scss)$": "identity-obj-proxy",
+    "\\.(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$":
+      "<rootDir>/src/tests/__mocks__/fileMock.js",
+  },
+  setupFiles: ["./src/setupTest.js"],
+  testPathIgnorePatterns: ["redux"],
+  coveragePathIgnorePatterns: ["redux"],
+};
```
Comment (GuevaraMarius) [inline jest.config.js]: Same here

File: .eslintcache [hunk 5]
```diff
@@ -0,0 +1 @@
...la/Desktop/phantom-frontend/incredible-phantom-frontend/src/componets/welcomeMessage.js":"4"},{"size":195,"mtime":1609229359713,"results":"5","hashOfConfig":"6"},{"size":291,"mtime":1609229359714,"results":"7","hashOfConfig":"6"},{"size":177,"mtime":1609229359713,"results":"8","hashOfConfig":"6"},{"size":342,"mtime":1609229359713,"results":"9","hashOfConfig":"6"},{"filePath":"10","messages":"11","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},"1axnbbs",{"filePath":"12","messages":"13","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},{"filePath":"14","messages":"15","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},{"filePath":"16","messages":"17","errorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},"/Users/andela/Desktop/phantom-frontend/incredible-phantom-frontend/src/index.js",[],"/Users/andela/Desktop/phantom-frontend/incredible-phantom-frontend/src/router/App.js",[],"/Users/andela/Desktop/phantom-frontend/incredible-phantom-frontend/src/pages/landingPage.js",[],"/Users/andela/Desktop/phantom-frontend/incredible-phantom-frontend/src/componets/welcomeMessage.js",[]]
```
Comment (GuevaraMarius) [inline .eslintcache]: Samuel you can stop tracking .eslintcache

File: .babelrc [hunk 6]
```diff
@@ -1,3 +1,5 @@
 {
```
Comment (GuevaraMarius) [inline .babelrc:1]: Remove empty lines

File: coverage/clover.xml [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+    <package name="router">
+      <metrics statements="2" coveredstatements="2" conditionals="0" coveredconditionals="0" methods="1" coveredmethods="1"/>
+      <file name="App.js" path="/Users/andela/Desktop/Projects/incredible-phantom-frontend/src/router/App.js">
+        <metrics statements="2" coveredstatements="2" conditionals="0" coveredconditionals="0" methods="1" coveredmethods="1"/>
+        <line num="6" count="1" type="stmt"/>
+        <line num="7" count="1" type="stmt"/>
+      </file>
+    </package>
+  </project>
+</coverage>
```
Comment (ceelogre) [inline coverage/clover.xml]: Do you need to track `coverage` files

**Review by ceelogre — CHANGES_REQUESTED** [review]
Component names typically start with a capital case.

File: coverage/clover.xml (hunk 7 above)
Comment (samuel776) [reply coverage/clover.xml]: thanks, and i worked on the requested changes

**Pull request closed without merging**